sdk/identity/azure-identity-extensions/src/main/java/com/azure/identity/extensions/implementation/template/AzureAuthenticationTemplate.java

```diff
@@ -60,8 +63,13 @@ public void init(Properties properties) {
             LOGGER.verbose("Initializing AzureAuthenticationTemplate.");
 
             if (getTokenCredentialProvider() == null) {
-                this.tokenCredentialProvider
-                    = TokenCredentialProvider.createDefault(new TokenCredentialProviderOptions(properties));
+                TokenCredentialProviderOptions options = new TokenCredentialProviderOptions(properties);
+                this.tokenCredentialProvider = TokenCredentialProvider.createDefault(options);
```

Comment:
Since the cache is shared between instances can we check it before making a new one?

Reply:
Yes, this is a caching idea, currently, our goal is to cache `TokenCredential`, and there is no plan to cache `TokenCredentialProvider` for the time being.